krux_marathon_api/marathonapi.py

```diff
@@ -86,7 +86,15 @@ def assign_config_data(self, config_file_data, marathon_app_result):
         marathon_app_result.health_checks            = config_file_data["health_checks"]
         marathon_app_result.labels                   = config_file_data["labels"]
         marathon_app_result.max_launch_delay_seconds = config_file_data["max_launch_delay_seconds"]
-        #marathon_app_result.ports = config_file_data["ports"]      # You cannot specify both ports and port definitions
+        ### If you define ports then blank the port definitions, but if you define
+        ### port_definitions they override ports, since port_definitions are more
+        ### specific (i.e. protocol and etc)
+        if "ports" in config_file_data:
+            marathon_app_result.ports                = config_file_data["ports"]
+            marathon_app_result.port_definitions     = []
+        if "port_definitions" in config_file_data:
```

Comment:
I think it is going to be more complex than this. If there is already a port definition on the server (which I think it makes automatically), then passing a port will cause an error. It won't change it.
I would make sure to test this part well to make sure it flips back and forth or is handled by an errors & logging.